core/domain/rte_component_registry.py

```diff
@@ -31,36 +30,15 @@ class Registry(object):
     def _refresh(cls):
         """Repopulate the registry."""
         cls._rte_components.clear()
-
-        # Assemble all paths to the RTE components.
-        extension_paths = [
-            component['dir'] for component in
-            feconf.ALLOWED_RTE_EXTENSIONS.values()]
-
-        # Crawl the directories and add new RTE component instances to the
-        # registry.
-        for loader, name, _ in pkgutil.iter_modules(path=extension_paths):
-            module = loader.find_module(name).load_module(name)
-            clazz = getattr(module, name)
-
-            ancestor_names = [
-                base_class.__name__ for base_class in clazz.__bases__]
-            if 'BaseRichTextComponent' in ancestor_names:
-                cls._rte_components[clazz.__name__] = clazz()
+        with open(feconf.RTE_EXTENSIONS_SPECS_PATH, 'r') as f:
+            cls._rte_components = constants.parse_json_from_js(f)
 
     @classmethod
     def get_all_rte_components(cls):
         """Get a list of instances of all custom RTE components."""
         if len(cls._rte_components) == 0:
```

Comment:
If I'm reading correctly, the data looks like a dict, but the docstring here and the use of len() indicate a list. Is this a mistake? If so let's fix this to say something like "if not cls._rte_components" and change the docstring to "Get a dict mapping RTE component IDs to their definitions." or similar.

Reply:
Done.